build/tools/esp32/esptool_py/Makefile

```diff
@@ -74,19 +74,32 @@ $(APP_BIN_UNSIGNED): $(APP_ELF) $(ESPTOOLPY_SRC)
 
 ESPTOOL_ALL_FLASH_ARGS += $(ESPTOOL_BOOTLOADER_FLASH_ARGS)
 
-ALL: flash romfs
+ESPTOOLPY_SUBMODULE := esptool_py/esptool
 
-OS:app-flash
+ALL: check-modules flash romfs
 
-ROMFS:romfs
+OS: check-modules app-flash
+
+ROMFS: check-modules romfs
+
+BOOTLOADER: check-modules bootloader
+
+ERASE_ALL: check-modules erase_flash
+
+check-modules:
+ifeq ($(ESPTOOLPY_SRC), $(wildcard $(ESPTOOLPY_SRC)))
+	@echo "$(ESPTOOLPY_SRC) exist"
```

Comment:
how about only echo when it's not exist.
and it's better to display the git command of getting submodule

Reply:
yes, now it could get submodule automatically. 
Thanks for your suggestion.